connect/src/main/scala/org/apache/spark/sql/catalyst/connect/connect.scala

```diff
@@ -0,0 +1,97 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.catalyst
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.catalyst.analysis.UnresolvedAlias
+import org.apache.spark.sql.catalyst.expressions.{Expression, NamedExpression}
+
+/**
+ * A collection of implicit conversions that create a DSL for constructing connect protos.
+ *
+ * {{{
+ *   scala> import org.apache.spark.sql.connect.plans.DslLogicalPlan
+ *
+ *   // Standard way to construct connect proto
+ *   scala> import org.apache.spark.connect.proto
+ *   scala> :paste
+ *   // Entering paste mode (ctrl-D to finish)
+ *   val connectTestRelation =
+ *    proto.Relation.newBuilder()
+ *     .setRead(
+ *       proto.Read.newBuilder()
+ *       .setNamedTable(proto.Read.NamedTable.newBuilder().addParts("student").build())
+ *       .build())
+ *     .build()
+ *  // Exiting paste mode, now interpreting.
+ *    connectTestRelation: org.apache.spark.connect.proto.Relation =
+ *     read {
+ *      named_table {
+ *        parts: "student"
+ *      }
+ *    }
+ *
+ *   // Now we can apply select on the proto relation above
+ *   scala> import org.apache.spark.sql.catalyst.analysis.UnresolvedAttribute
+ *   scala> connectTestRelation.select(UnresolvedAttribute(Seq("id")))
+ *   res14: org.apache.spark.connect.proto.Relation =
+ *    project {
+ *      input {
+ *        read {
+ *          named_table {
+ *            parts: "student"
+ *          }
+ *        }
+ *      }
+ *      expressions {
+ *        unresolved_attribute {
+ *          parts: "id"
+ *        }
+ *      }
+ *    }
+ *
+ * }}}
+ *
+ */
+package object connect {
+
+  object plans { // scalastyle:ignore
+    implicit class DslLogicalPlan(val logicalPlan: proto.Relation) {
+      def select(exprs: Expression*): proto.Relation = {
+        val namedExpressions = exprs.map {
+          case e: NamedExpression => e
+          case e => UnresolvedAlias(e)
+        }
+
+        proto.Relation.newBuilder().setProject(
+          proto.Project.newBuilder()
+            .setInput(logicalPlan)
+            .addExpressions(
+              proto.Expression.newBuilder()
+                .setUnresolvedAttribute(
```

Comment:
Sorry I'm still a bit confused about this. How are we going to test more expressions in the future? Are we going to put a catalyst `Expression` to proto expression converter here?

Reply:
I am not familiar with existing analyzer DSL enough bu
1. This DSL matches with the analyzer DSL API. So the inputs will be the same.
2. Then I think yes, we gonna have more conversions. Function -> UnresolvedFunciton, Column -> UnresolvedAttribute, Literal -> proto defined literal, etc.

Reply:
I see why you might be confused: 

the code here is definitely not complete (of course it is hard to make it complete from the beginning).  

I plan to expand this when we start to implement one API after another. 

Basically when we start to check the `Projection` support, we then test every type of the expression for a `select` on a relation. After that this part will be complete (I hope).

Reply:
Added the DSL part for expressions.